modules/other/vm_with_power_automate_desktop/main.tf

```diff
@@ -0,0 +1,133 @@
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.48"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.5"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.1"
+    }
+    http = {
+      source  = "hashicorp/http"
+      version = "~> 2.1"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = "~> 2.1"
+    }
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_public_ip" "public_ip" {
+  count = var.enable_public_access ? 1 : 0
+
+  name                = "public-ip-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  allocation_method   = "Dynamic"
+}
+
+resource "azurerm_network_interface" "network_interface" {
+  name                = "nic-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.subnet_id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.enable_public_access ? azurerm_public_ip.public_ip[0].id : null
+  }
+}
+
+resource "azurerm_network_security_group" "nsg" {
+  count = var.enable_public_access ? 1 : 0
+
+  name                = "nsg-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  security_rule {
+    name                       = "Allow_RDP"
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "3389"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+
+}
+
+resource "azurerm_network_interface_security_group_association" "association" {
+  count = var.enable_public_access ? 1 : 0
+
+  network_interface_id      = azurerm_network_interface.network_interface.id
+  network_security_group_id = azurerm_network_security_group.nsg[0].id
+}
+
+resource "random_password" "vm_user_password" {
+  length           = 16
+  special          = true
+  override_special = "_%@"
```

Comment:
Is there a reason why we are overriding the special characters to only use the characters displayed?

Reply:
Not really, changed it